### DISCUSSION TITLE
Skip repositories by ID

### DISCUSSION THREAD

**Discussion by Katsute** [lowlighter/metrics#729] [Ideas]
Currently the repository skip parameter only accepts `{owner}/{repo}` or `{repo}`, would it be possible to also skip by using the repository ID? When I calculate statistics I would like to include some, but not all of my private repositories, but I don't want to expose the names of the skipped repositories.

**Comment by lowlighter** [comment]
I think the best way would be to use a repository secret instead, which would also censor every mention of your repositories in your logs
```yaml
repositories_skipped: public_repo1, ${{ secrets.SKIPPED_REPOSITORIES }} 
# with SKIPPED_REPOSITORIES: "private_repo1, private_repo2"
```

**Reply by Katsute** [reply]
I thought about that, but for some weird reason the names are still exposed in the log

**Reply by lowlighter** [reply]
It may depend on how these are displayed, iirc GitHub only censor exact matches

**Reply by Katsute** [reply]
Copying the exact output from the log seems to have fixed the masking, it would still be nice to have the IDs as an option though

**Reply by lowlighter** [reply]
Afaik I don't think repositories ids are easily accessible so it seems to be more of an edge case 

Supporting repositories ids would need to patch each plugin which use a filter list so they can fetch ids and compare them. It seems a lot of work for an addition that may not be used often and I guess secrets works fine enough

I'll archive this discussion, thanks again for your feedback 👍
